extension/opampextension/opamp_agent.go

```diff
@@ -0,0 +1,205 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opampextension
+
+import (
+	"context"
+	"os"
+	"runtime"
+
+	"github.com/oklog/ulid/v2"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opamp-go/client"
+	"github.com/open-telemetry/opamp-go/client/types"
+	"github.com/open-telemetry/opamp-go/protobufs"
+)
+
+// TODO: Replace with https://github.com/open-telemetry/opentelemetry-collector/issues/6596
+const localConfig = `
+exporters:
+  otlp:
+    endpoint: localhost:1111
+receivers:
+  otlp:
+    protocols:
+      grpc: {}
+      http: {}
+service:
+  pipelines:
+    traces:
+      receivers: [otlp]
+      processors: []
+      exporters: [otlp]
+`
+
+type opampAgent struct {
+	cfg    *Config
+	logger *zap.Logger
+
+	agentType    string
```

Comment:
This will ultimately just come from the `service.name` attribute that the Collector uses to identify itself. Usually this will be `io.opentelemetry.collector`, but in the future could be `com.vendor.agent` or something similar. This field and related fields to identify the Collector can likely be removed and replaced with a map of the Collector's resource attributes defined in [`component.TelemetrySettings`](https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/telemetry.go#L31).